Syndication/AzureStack.MarketplaceSyndication.psm1

```diff
@@ -90,165 +83,1086 @@ function Sync-AzSOfflineMarketplaceItem {
-                    $vhdSource = $downloadDetails.properties.osDiskImage.sourceBlobSasUri
-                    If ([string]::IsNullOrEmpty($vhdsource)) {exit} else {
-                        $FileExists = Test-Path "$destination\$vhdName.vhd" 
-                        If ($FileExists -eq $true) {Remove-Item "$destination\$vhdName.vhd" -force} else {
-                            New-Item "$destination\$vhdName.vhd" 
+                    If ($downloadConfirmation -eq 'Y') {
+                        $checktool= Test-Path "C:\Program Files (x86)\Microsoft SDKs\Azure\AzCopy\AzCopy.exe"
+                        If ($checktool -eq $true){
+                            DownloadMarketplaceProduct -Source $vhdsource -Destination $vhddestination -ProductName "$vhdName.vhd" -PremiumDownload -MaxRetry 2
+                            "$productFolder\$vhdName.vhd"|out-file "$productFolder\$azpkgName.txt" -Append
                         }
-                        $vhdDestination = "$destination\$vhdName.vhd"
-
-                        #Select Premium Download
-            $a = new-object -comobject wscript.shell
-            $intAnswer = $a.popup("Would you like to use Premium download? This requires Azure Storage Tools to be installed", `
-                    0, "Premium Download", 4)
-        If ($intAnswer -eq 6) {
-            ($checktool= test-path 'C:\Program Files (x86)\Microsoft SDKs\Azure\AzCopy\AzCopy.exe' )
-            If ($checktool -eq $true){        
-            & 'C:\Program Files (x86)\Microsoft SDKs\Azure\AzCopy\AzCopy.exe' /Source:$vhdsource /Dest:$vhddestination /Y
-            $a.popup("$vhdName.vhd has been downloaded")
-            "$destination\$vhdName.vhd"|out-file "$destination\$azpkgName.txt" -Append
-            }
-            else{
-                $a.popup("Please install Azure Storage Tools first,canceling")
+                        else{
+                            Write-Host "Please install Azure Storage Tools first,canceling"
+                        }
+                    }else
+                    {
+                        DownloadMarketplaceProduct -Source $vhdsource -Destination $vhddestination -ProductName "$vhdName.vhd" -MaxRetry 2
+                    }
+                    Write-Host "$vhdName.vhd has been downloaded"
+                    "$productFolder\$vhdName.vhd"|out-file "$productFolder\$azpkgName.txt" -Append
+                }
             }
-        }else
-        {(New-Object System.Net.WebClient).DownloadFile("$vhdsource",$vhddestination) }
-        $a.popup("$vhdName.vhd has been downloaded")
-        "$destination\$vhdName.vhd"|out-file "$destination\$azpkgName.txt" -Append
+            'virtualMachineExtension' {
+                # download zip
+                $zipName = $productDetails.properties.galleryItemIdentity
+                $zipsource = $downloadDetails.properties.sourceBlob.uri
+                If ([string]::IsNullOrEmpty($zipsource)) {exit} else {
```

Comment:
there are a lot of nested if statements which makes this code hard to read and follow. at a minimum ensure you are using consistent formatting with the else blocks on new lines, etc.